assets/app/scripts/controllers/edit/autoscaler.js

```diff
@@ -0,0 +1,199 @@
+      return;
+    }
+
+    $scope.kind = $routeParams.kind;
+    $scope.name = $routeParams.name;
+    if ($routeParams.kind === "HorizontalPodAutoscaler") {
+      // Wait for the HPA data to load before enabling the form controls.
+      // This is only necessary when editing an existing HPA.
+      $scope.disableInputs = true;
+    } else {
+      $scope.targetKind = $routeParams.kind;
+      $scope.targetName = $routeParams.name;
+    }
+
+    $scope.autoscaling = {
+      name: $scope.name,
+      labels: {}
+    };
+
+    // Warn if metrics aren't configured when setting autoscaling options.
+    MetricsService.isAvailable().then(function(available) {
+      $scope.metricsWarning = !available;
+    });
+
+    $scope.alerts = {};
+
+    // More breadcrumbs inserted later as data loads.
+    $scope.breadcrumbs = [{
+      title: $routeParams.project,
+      link: "project/" + $routeParams.project
+    }, {
+      title: "Deployments",
+      link: "project/" + $routeParams.project + "/browse/deployments"
+    }, {
+      title: "Autoscale"
+    }];
+
+    var getErrorDetails = $filter('getErrorDetails');
+
+    var displayError = function(errorMessage, result) {
+      $scope.alerts['autoscaling'] = {
+        type: "error",
+        message: errorMessage,
+        details: getErrorDetails(result)
+      };
+    };
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        // Update project breadcrumb with display name.
+        $scope.breadcrumbs[0].title = $filter('displayName')(project);
+        $scope.project = project;
+
+        var createHPA = function() {
+          $scope.disableInputs = true;
+          var hpa = {
+            apiVersion: "extensions/v1beta1",
+            kind: "HorizontalPodAutoscaler",
+            metadata: {
+              name: $scope.autoscaling.name,
+              labels: $scope.autoscaling.labels
+            },
+            spec: {
+              scaleRef: {
+                kind: $routeParams.kind,
+                name: $routeParams.name,
+                apiVersion: "extensions/v1beta1",
+                subresource: "scale"
+              },
+              minReplicas: $scope.autoscaling.minReplicas,
+              maxReplicas: $scope.autoscaling.maxReplicas,
+              cpuUtilization: {
+                targetPercentage: $scope.autoscaling.targetCPU || $scope.autoscaling.defaultTargetCPU
+              }
+            }
+          };
+
+          DataService.create({
+            resource: 'horizontalpodautoscalers',
+            group: 'extensions'
+          }, null, hpa, context)
+            .then(function() { // Success
+              // Return to the previous page
+              $window.history.back();
+            }, function(result) { // Failure
+              $scope.disableInputs = false;
+              displayError('An error occurred creating the horizontal pod autoscaler', result);
```

Comment:
yeah period or colon, needs one of them. You have a space at the end of
your other call to displayError so just make them consistent whichever way
you go.

On Thu, Mar 24, 2016 at 1:31 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/controllers/edit/autoscaler.js
> <https://github.com/jwforres/origin/pull/24#discussion_r57355058>:
>
> > +              cpuUtilization: {
> > +                targetPercentage: $scope.autoscaling.targetCPU || $scope.autoscaling.defaultTargetCPU
> > +              }
> > +            }
> > +          };
> > +
> > +          DataService.create({
> > +            resource: 'horizontalpodautoscalers',
> > +            group: 'extensions'
> > +          }, null, hpa, context)
> > +            .then(function() { // Success
> > +              // Return to the previous page
> > +              $window.history.back();
> > +            }, function(result) { // Failure
> > +              $scope.disableInputs = false;
> > +              displayError('An error occurred creating the horizontal pod autoscaler', result);
>
> Space or period? It looks OK without the space, but probably needs the
> period?
>
> [image: openshift_web_console]
> <https://cloud.githubusercontent.com/assets/1167259/14025460/943571d6-f1c4-11e5-9c5c-bbbb48f2f46e.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jwforres/origin/pull/24/files/df39d33b4c94677b6b0dbbabddca69acfd858492#r57355058>
>